pkg/kubectl/rollback.go

```diff
@@ -221,9 +224,10 @@ func (r *DaemonSetRollbacker) Rollback(obj runtime.Object, updatedAnnotations ma
 	if !ok {
 		return "", fmt.Errorf("passed object is not a DaemonSet: %#v", obj)
 	}
-	versionedExtensionsClient := versionedExtensionsClientV1beta1(r.c)
 	versionedAppsClient := versionedAppsClientV1beta1(r.c)
-	versionedDS, allHistory, err := controlledHistories(versionedExtensionsClient, versionedAppsClient, ds.Namespace, ds.Name)
+	versionedExtensionsClient := versionedExtensionsClientV1beta1(r.c)
+	versionedObj, allHistory, err := controlledHistories(versionedAppsClient, versionedExtensionsClient, ds.Namespace, ds.Name, "DaemonSet")
+	versionedDS := versionedObj.(*externalextensions.DaemonSet)
```

Comment:
Noted! Thanks for pointing it out - this is very important as i use type assertion frequently.